packages/react-components/react-motion/library/src/hooks/useAnimateAtoms.ts

```diff
@@ -1,8 +1,11 @@
 import * as React from 'react';
 import type { AnimationHandle, AtomMotion } from '../types';
 
+// eslint-disable-next-line no-restricted-globals
+const win = typeof window !== undefined ? window : undefined;
+
 // Heads up! "Element." is a side-effect for minifiers, should be kept as IIFE to avoid leaking after minification.
-const SUPPORTS_WEB_ANIMATIONS = /*@__PURE__*/ (() => typeof Element.prototype.animate === 'function')();
+const SUPPORTS_WEB_ANIMATIONS = /*@__PURE__*/ (() => win && typeof win.Element.prototype.animate === 'function')();
```

Comment:
I was wondering why lint didn't trigger error on this. 

`Element` is missing from our definition list 🚨 ( is that on purpose @spmonahan  ? )

<img width="335" alt="image" src="https://github.com/microsoft/fluentui/assets/1223799/6538dd4c-33ef-473b-afe7-0452338cddb2">

⬇

<img width="1244" alt="image" src="https://github.com/microsoft/fluentui/assets/1223799/6f70783d-2550-42fa-83e9-2827becb4f5e">


can we add it to eslint setup ?

Reply:
It's because the lint rule flags both types and values equally. We need to implement a type-aware version of the `no-restricted-globals` plugin. See #31286

Current state with `no-restricted-globals` is:

```ts
// We want to allow this but the lint rule will error on it
let myElement: Element;

// And we don't want this
myElement = new Element(); // NOTE: this will throw anyway, just for illustration purposes
```

Reply:
yeah I know, but somehow in my memory was that we temporarily use eslint-disable inline pragmas for those type symbols. ok that's not the case. ty